getgather/mcp/brand/goodreads.py

```diff
@@ -1,32 +1,39 @@
-import os
 from typing import Any
 
 from fastmcp import Context
+from patchright.async_api import Page
 
-from getgather.distill import load_distillation_patterns, run_distillation_loop
 from getgather.mcp.agent import run_agent_for_brand
+from getgather.mcp.dpage import dpage_mcp_tool
 from getgather.mcp.registry import BrandMCPBase
-from getgather.mcp.shared import get_mcp_browser_profile
 from getgather.mcp.stagehand_agent import (
     run_stagehand_agent,
 )
 
 goodreads_mcp = BrandMCPBase(brand_id="goodreads", name="Goodreads MCP")
 
 
-@goodreads_mcp.tool(tags={"private"})
+@goodreads_mcp.tool
+async def get_username() -> dict[str, Any]:
+    async def callback(page: Page) -> dict[str, Any]:
+        await page.click("a:has-text('My Books')")
+        return {"url": page.url}
+
+    return await dpage_mcp_tool(
+        "https://www.goodreads.com/recommendations",
+        "url",
```

Comment:
the second argument is irrelevant here. Since the value being returned is coming from line 20 above. Will refactor this when we split dpage_mcp_tool into its own function